homeassistant/components/cloud/__init__.py

```diff
@@ -252,11 +263,22 @@ async def _on_connect():
             Platform.TTS, DOMAIN, {}, config
         )
 
+        async_dispatcher_send(
```

Comment:
We can't use the dispatcher if we want to allow other integrations to subscribe to the cloud state. The dispatcher is only for internal communication within an integration or within one helper.

Reply:
What's the best practice for a proper API function then?

Events are not wanted since they are exposed to user and recorded, which is not useful in this case.
Duplicating what's nicely implemented by the dispatcher will either lack needed functionality (with the current commit, it's not possible to unsubscribe which is not OK), or require reinventing the dispatcher wheel.

Reply:
It would be ok to wrap the dispatcher but it's not ok to use the signal string + the dispatcher send method as interface.

Reply:
OK, makes sense

Reply:
I'd expect something like this:

```py
async_subscribe_cloud_state(
    hass: HomeAssistant, callback: Callable[[CloudConnectionState], None]
) -> CALLBACK_TYPE:
    return = async_dispatcher_connect(hass, SIGNAL, callback)
```